### PR TITLE
Run module doc tests

### PR DESCRIPTION
**Pull request by killercup**
Just a small thing that makes running doc tests possible/useful. A follow-up PR could add some tests that assert some matches and/or add docs to filter methods.

### CODE REVIEW

**Comment by Gilnaa** [comment]
Great! Didn't know you could hide lines in rustdoc

File: src/lib.rs [hunk 1]
```diff
@@ -31,55 +31,73 @@
 //!
 //! ## Finding image files in the current directory.
 //!
-//! ```no_run
+//! ```rust
 //! extern crate globwalk;
+//! # include!("doctests.rs");
```
Comment (epage) [inline src/lib.rs:36]: Huh, that is a neat idea for extra test code!  I feel like the this should be a tip documented somewhere.

Comment (killercup) [reply src/lib.rs:36]: [Oh but it is documented as pro tip!](https://killercup.github.io/rustfest-idiomatic-libs/index.html#/4/3) ;)

Comment (epage) [reply src/lib.rs:36]: Glossed over that last time I looked at your presentation.  Ideally we find permanent central homes for these things so people don't have to dig through all of the past presentations / blog posts and hope its still the latest.

**Pull request merged**